### DISCUSSION TITLE
Should we copy the excalidraw plugin and have CookLang Markdown (`.cook.md`) files?

### DISCUSSION THREAD

**Discussion by deathau** [cooklang/cooklang-obsidian#10] [Ideas]
## Pros:
- Will go into the cache, allowing base Obsidian functionality such as:
    - Appearing on graph view
    - Appearing in search
    - Embedding in markdown files
    - Appearing in dataview queries

## Cons:
- A bit of development work to check for specific metadata and overwrite the default Markdown views
- Not true CookLang files, are actually `.cook.md`, therefore incompatible with other CookLang software.

Feel free to make any other points you can think of.

**Comment by kometenstaub** [comment]
You could add an export option from `.cook.md` to `.cook`. Maybe one option to do it on a per file basis, and one to export all `.cook.md` files.

**Reply by deathau** [reply]
This is a good point. And if I go down the `.cook.md` route, I will definitely have to include import/export

**Comment by claremacrae** [comment]
It feels cleaner to call them .cook files - reasons include:

- other software may start supporting .cook files, and it will cause needless complications if two different systems use different extensions (Oh, I see I'm agreeing with your second point above)
- on Windows, the default is to hide file extensions - many users don't know this, and leave it turned off - in which case, I believe they will see .cook.md files named as .cook files - which will also be confusing

**Reply by deathau** [reply]
Yeah, these are good points, and exactly why I'm leaning away from this currently.
But as it stands, there's no other way to get `.cook` files into the cache.

**Reply by claremacrae** [reply]
What are the consequences of not being in the cache?

**Reply by RichSteele** [reply]
Besides not appearing in dataview queries the bigger thing is that normal Obsidian text queries won't find non-md files. So, I can find a specific recipe using file: but not all recipes containing "egg", for example.

**Reply by deathau** [reply]
Also embeds. Sometimes it would be nice to embed recipies in specific pages.

**Reply by claremacrae** [reply]
Tricky…

**Reply by dpickett** [reply]
the other downside is you can't use other plug-ins like vim mode
